AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -5265,39 +5274,40 @@ function Compress-AzResourceId {
     }
 }
 
-function Request-ConnectAzureAD {
+function Request-ConnectMsGraph {
     <#
     .SYNOPSIS
-    Connect to an Azure AD tenant using the AzureAD cmdlets.
+    Connect to an Azure AD tenant using the MsGraph cmdlets.
     .DESCRIPTION
-    Correctly import the AzureAD module for your PowerShell version and then sign in using the same tenant is the currently signed in Az user. This wrapper is necessary as 1. AzureAD is not directly compatible with PowerShell 6 (though this can be achieved through the WindowsCompatibility module), and 2. AzureAD doesn't necessarily log you into the same tenant as the Az cmdlets according to their documentation (although it's not clear when it doesn't).
+    Correctly import the MsGraph module for your PowerShell version and then sign in using the same tenant is the currently signed in Az user. This wrapper is necessary as 1. AzureAD is not directly compatible with PowerShell 6 (though this can be achieved through the WindowsCompatibility module), and 2. AzureAD doesn't necessarily log you into the same tenant as the Az cmdlets according to their documentation (although it's not clear when it doesn't).
     .EXAMPLE
-    Request-ConnectAzureAD
+    Request-ConnectMsGraph
     #>
 
     [CmdletBinding()]
     param()
 
     Assert-IsWindows
-    Request-AzureADModule
+    Request-MSGraphModule
 
-    $aadModule = Get-Module | Where-Object { $_.Name -like "AzureAD" }
-    if ($null -eq $aadModule) {
+    $msGraphModule = Get-Module | Where-Object { $_.Name -like "Microsoft.Graph" }
+    if ($null -eq $msGraphModule) {
         if ($PSVersionTable.PSVersion -ge [Version]::new(6,0,0,0)) {
-            Import-WinModule -Name AzureAD -Verbose:$false
+            Import-WinModule -Name Microsoft.Graph -Verbose:$false
         } else {
-            Import-Module -Name AzureAD
+            Import-Module -Name Microsoft.Graph
         }
     }
 
     try {
-        Get-AzureADTenantDetail -ErrorAction Stop | Out-Null
+        Get-MgOrganization -ErrorAction Stop | Out-Null
     } catch {
         $context = Get-AzContext
-        Connect-AzureAD `
-                -TenantId $context.Tenant.Id `
-                -AccountId $context.Account.Id `
-                -AzureEnvironmentName $context.Environment.Name | `
+        Connect-MgGraph `
+                -TenantId $context.TenantId `
+                # we will need to read user/group to get sid/oid, and domain to get domain information
+                -Scopes "User.Read.All,GroupMember.Read.All,Domain.Read.All" `

```

Comment:
Rather than hardcoding the scopes here, should we take them as parameters? I'm just thinking that some commands may require fewer access rights, and it could be an issue to request too many if the user only needs to run a single command but not another

Reply:
Updated

Reply:
Does PowerShell or MS Graph support both "a,b,c" and "a", "b", "c" list parameters? Just noticed we have both styles

Reply:
the new code does not have this part, but I tried with "a,b,c", it worked on my side. from the official doc, we should use "a","b","c", you can see the new version all changed to "a","b","c".